src/PlaybackLooping.h

```diff
@@ -0,0 +1,101 @@
+/*!********************************************************************
+*
+ Audacity: A Digital Audio Editor
+
+ PlaybackLooping.h
+
+ Anton Gerasimov
+
+ **********************************************************************/
+
+#pragma once
+
+#include <wx/dc.h>
+
+#include "ViewInfo.h"
+#include "TrackPanelCell.h"
+#include "UIHandle.h"
+
+//! Playback looping service (processor).
+class AUDACITY_DLL_API PlaybackLooping
+{
+public:
+    PlaybackLooping();
+
+    static PlaybackLooping& GetInstance();
+
+    bool mLoopingEnabled{ false };
+    bool IsLoopingEnabled() const;
+    void EnableLooping(bool enable);
+
+    enum class ProcessingStates
+    {
+        kIdle,
+        kEnterLooping,
+        kLooping,
+        kDragging
+    };
+    ProcessingStates mProcessingState;
+
+    enum class ProcessingEvents
+    {
+        kNone,
+        kClick,
+        kDrag,
+        kRelease,
+        kPreview,
+        kCancel
+    };
+
+    bool LoopingProcessor(AudacityProject* pProject, wxCoord& mousePosX, ProcessingEvents processingEvents);
+
+    bool IsProcessing() const;
+
+    PlayRegion mOldPlayRegion;
+    PlayRegion mLoopedPlayRegion;
```

Comment:
Why are there new play region objects?  I expected changes in the display of the existing play region object.  I expected replacement of the double headed red arrow that shows the locked play region.  Instead I see both when I build.  And the new rectangle only displays, but doesn't affect what the play button does.